app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -9,9 +9,11 @@ class Course::Assessment::Submission::SubmissionsController < \
   before_action :check_password, only: [:edit, :update]
   before_action :load_or_create_answers, only: [:edit, :update]
   before_action :check_zombie_jobs, only: [:edit]
+  before_action :create_submission_questions, only: :create
```

Comment:
this should be similar to `load_or_create_answers`, and need to be called on both `edit` and `update`